src/core/server/saved_objects/service/lib/aggregations/aggs_types/metrics_aggs.ts

```diff
@@ -68,7 +68,7 @@ export const metricsAggsSchemas: Record<string, ObjectType> = {
     stored_fields: s.maybe(s.oneOf([s.string(), s.arrayOf(s.string())])),
     from: s.maybe(s.number()),
     size: s.maybe(s.number()),
-    sort: s.maybe(s.oneOf([s.literal('asc'), s.literal('desc')])),
+    sort: s.maybe(s.any()),
```

Comment:
Sorry, I would really like to avoid `any` in the aggregation schemas. I'm very fine changing the `sort` schema to get closer to the real type (it do not need to be exhaustive here, only to match your needs), but the whole goal of the aggs validation is to not allow arbitrary values.

```ts
export type Sort = SortCombinations | Array<SortCombinations>

export type SortCombinations = Field | SortContainer | SortOrder

export type SortOrder = [string, string];

export interface Field {
  name: string;
  type: string;
  esTypes?: string[];
  aggregatable: boolean;
  filterable: boolean;
  searchable: boolean;
  subType?: FieldSubType;
}

export interface SortContainerKeys {
  _score?: ScoreSort
  _doc?: ScoreSort
  _geo_distance?: GeoDistanceSort
  _script?: ScriptSort
}
export type SortContainer = SortContainerKeys |
    { [property: string]: FieldSort | SortOrder }
```

From your code, it seems you're using the `SortContainer` format

I would introduce a `SortSchema`, and also use it in other aggregations where we have a `sort` options, e.g `terms`.



```ts
// note: these schemas are not exhaustive. See the `Sort` type of `@elastic/elasticsearch` if you need to enhance it.
const sortOrderSchema = s.oneOf([s.literal('asc'), s.literal('desc'), s.literal('_doc')]);
const sortContainerSchema = s.recordOf(s.string(), sortOrderSchema);
const sortCombinationsSchema = s.oneOf([sortContainerSchema, sortOrderSchema])
const sortSchema = s.oneOf([sortCombinationsSchema, s.arrayOf(sortCombinationsSchema)])
```

You can create a `src/core/server/saved_objects/service/lib/aggregations/aggs_types/common_schemas.ts` file for that.

Reply:
No problem, thanks for the feedback! I added the file you suggested with some of the reusable sort schemas and used them to enhance the `terms.order` and top hits sort validation.